src/wp-includes/user.php

```diff
@@ -1849,9 +1849,18 @@ function update_user_caches( $user ) {
 	}
 
 	wp_cache_add( $user->ID, $user, 'users' );
-	wp_cache_add( $user->user_login, $user->ID, 'userlogins' );
-	wp_cache_add( $user->user_email, $user->ID, 'useremail' );
-	wp_cache_add( $user->user_nicename, $user->ID, 'userslugs' );
+
+	if ( ! empty( $user->user_login ) ) {
+		wp_cache_add( $user->user_login, $user->ID, 'userlogins' );
+	}
+
+	if ( ! empty( $user->user_email ) ) {
+		wp_cache_add( $user->user_email, $user->ID, 'useremail' );
+	}
+
+	if ( ! empty( $user->user_nicename ) ) {
+		wp_cache_add( $user->user_nicename, $user->ID, 'userslugs' );
+	}
```

Comment:
5 test cases were failing because `user_email` was an empty string. This could lead to terrible behavior in the real world 😬

<details>
  <summary>Failed tests...</summary>
<pre><code>
1) Tests_User::test_wp_insert_user_with_meta
Unexpected incorrect usage notice for add.
Cache key must not be an empty string. (This message was added in version 6.1.0.)
Failed asserting that an array is empty.
/var/www/tests/phpunit/includes/abstract-testcase.php:589
/var/www/tests/phpunit/includes/abstract-testcase.php:601
/var/www/vendor/yoast/phpunit-polyfills/src/TestCases/TestCasePHPUnitLte7.php:94
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:52
2) Tests_User_GetUserCount::test_get_user_count_on_different_network
Unexpected incorrect usage notice for add.
Cache key must not be an empty string. (This message was added in version 6.1.0.)
Unexpected incorrect usage notice for delete.
Cache key must not be an empty string. (This message was added in version 6.1.0.)
Failed asserting that an array is empty.
/var/www/tests/phpunit/includes/abstract-testcase.php:589
/var/www/tests/phpunit/includes/abstract-testcase.php:601
/var/www/vendor/yoast/phpunit-polyfills/src/TestCases/TestCasePHPUnitLte7.php:94
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:52
3) Tests_User_GetUserCount::test_enable_live_network_user_counts_filter
Unexpected incorrect usage notice for add.
Cache key must not be an empty string. (This message was added in version 6.1.0.)
Unexpected incorrect usage notice for delete.
Cache key must not be an empty string. (This message was added in version 6.1.0.)
Failed asserting that an array is empty.
/var/www/tests/phpunit/includes/abstract-testcase.php:589
/var/www/tests/phpunit/includes/abstract-testcase.php:601
/var/www/vendor/yoast/phpunit-polyfills/src/TestCases/TestCasePHPUnitLte7.php:94
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:52
4) Tests_User_GetUserCount::test_get_user_count_update_on_delete_multisite
Unexpected incorrect usage notice for add.
Cache key must not be an empty string. (This message was added in version 6.1.0.)
Unexpected incorrect usage notice for delete.
Cache key must not be an empty string. (This message was added in version 6.1.0.)
Failed asserting that an array is empty.
/var/www/tests/phpunit/includes/abstract-testcase.php:589
/var/www/tests/phpunit/includes/abstract-testcase.php:601
/var/www/vendor/yoast/phpunit-polyfills/src/TestCases/TestCasePHPUnitLte7.php:94
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:52
5) Tests_User_Slashes::test_wp_insert_user
Unexpected incorrect usage notice for add.
Cache key must not be an empty string. (This message was added in version 6.1.0.)
Unexpected incorrect usage notice for delete.
Cache key must not be an empty string. (This message was added in version 6.1.0.)
Failed asserting that an array is empty.
/var/www/tests/phpunit/includes/abstract-testcase.php:589
/var/www/tests/phpunit/includes/abstract-testcase.php:601
/var/www/vendor/yoast/phpunit-polyfills/src/TestCases/TestCasePHPUnitLte7.php:94
phpvfscomposer:///var/www/vendor/phpunit/phpunit/phpunit:52
</code></pre>
</details>

Reply:
Technically users with `'0'` as their `user_login` or `user_nicename` would never be cached with these additions.